app/assets/javascripts/discourse-common/lib/icon-library.js.es6

```diff
@@ -68,58 +236,89 @@ export function registerIconRenderer(renderer) {
   _renderers.unshift(renderer);
 }
 
-// Support for font awesome icons
-function faClasses(icon, params) {
-  let classNames = `fa fa-${icon.replacementId || icon.id} d-icon d-icon-${
-    icon.id
-  }`;
+function iconClasses(icon, params) {
+  // "notification." is invalid syntax for classes, use replacement instead
+  let dClass =
+    icon.replacementId && icon.id.indexOf("notification.") > -1
+      ? icon.replacementId
+      : icon.id;
 
-  if (params) {
-    if (params.modifier) {
-      classNames += " fa-" + params.modifier;
-    }
-    if (params["class"]) {
-      classNames += " " + params["class"];
-    }
+  let classNames = `fa d-icon d-icon-${dClass} svg-icon`;
+
+  if (params && params["class"]) {
+    classNames += " " + params["class"];
   }
+
   return classNames;
 }
 
+function warnIfMissing(id) {
+  if (
+    typeof Discourse !== "undefined" &&
+    Discourse.Environment === "development" &&
+    Discourse.SvgIconList &&
+    Discourse.SvgIconList.indexOf(id) === -1
+  ) {
+    console.warn(`The icon "${id}" is missing from the SVG subset.`);
+  }
+}
+
+function warnIfDeprecated(oldId, newId) {
+  if (
+    typeof Discourse !== "undefined" &&
+    Discourse.Environment === "development" &&
+    !Ember.testing
+  ) {
+    deprecated(`Icon "${oldId}" is now "${newId}".`);
+  }
+}
+
+function handleIconId(icon) {
+  let id = icon.replacementId || icon.id || "";
+
+  if (fa4Replacements.hasOwnProperty(id)) {
+    warnIfDeprecated(id, fa4Replacements[id]);
+    id = fa4Replacements[id];
+  } else if (id.substr(id.length - 2) === "-o") {
+    let newId = "far-" + id.replace("-o", "");
+    warnIfDeprecated(id, newId);
+    id = newId;
+  }
+
+  // TODO: clean up "thumbtack unpinned" at source instead of here
+  id = id.replace(" unpinned", "");
+
+  warnIfMissing(id);
+  return id;
+}
+
 // default resolver is font awesome
 registerIconRenderer({
   name: "font-awesome",
 
-  string(icon, params) {
-    let tagName = params.tagName || "i";
-    let html = `<${tagName} class='${faClasses(icon, params)}'`;
-    if (params.title) {
```

Comment:
SVG elements can't have a title attribute. But does this mean we're regressing on accessibility here?